src/components/addition/DetailedCard.tsx

```diff
@@ -0,0 +1,51 @@
+import { ProductData, ProductsResponse } from '../../interfaces';
+import { useRouter } from 'next/router';
+
+function DetailedCard({ data }: ProductsResponse) {
+  const router = useRouter();
+  const { page, limit, search, details } = router.query;
+  const targetProductObj = data?.results?.find(
```

Comment:
In previous task you have to make a separate request for details... You are cheater!